### DISCUSSION TITLE
support for compiling cuda using clang

### DISCUSSION THREAD

**Discussion by volatileAntiEntropy** [xmake-io/xmake#2701] [Q&A]
I'm wondering if there is any equivalency to `add_cucodegens("native")` for the `clang` tool chain, or more generally, is there any possibility that we can have a tool chain that uses `clang` and one pass compilation for `cuda` code?

**Comment by waruqi** [comment]
not supported yet, but there a workaround solution. https://github.com/xmake-io/xmake/issues/2305

**Reply by volatileAntiEntropy** [reply]
Well, I get warnings like `warning: add_cuflags("--cuda-gpu-arch=sm_86") is ignored, please pass {force = true} or call set_policy("check.auto_ignore_flags", false) if you want to set it.`. However, if I do `set_policy("check.auto_ignore_flags", false)`, nothing changes, I still get that warning.
